graphistry/feature_utils.py

```diff
@@ -2802,20 +2803,22 @@ def featurize_or_get_nodes_dataframe_if_X_is_None(
         """helper method gets node feature and target matrix if X, y are not specified. if X, y are specified will set them as `_node_target` and `_node_target` attributes
         """
 
-        res = self.bind()
+        res = cast('FeatureMixin', self.bind())
 
         if not reuse_if_existing:  # will cause re-featurization
             res._node_features = None
             res._node_target = None
 
         if reuse_if_existing and res._node_features is not None:
             logger.info('-Reusing Existing Node Featurization')
-            return res._node_features, res._node_target, res
+            # NOTE: Wasn't sure if " res._node_target is not None" ought to have been a condition for this code path
+            node_target = res._node_target if res._node_target is not None else pd.DataFrame()
```

Comment:
i missed it on the first pass of these, but i wonder if for these `df.empty` `y` stubs, we need `len(y.index) == len(X.index)`, even if no columns...